components/script/dom/bindings/utils.rs

```diff
@@ -178,28 +181,46 @@ pub unsafe fn get_property_on_prototype(
 
 /// Get an array index from the given `jsid`. Returns `None` if the given
 /// `jsid` is not an integer.
-pub fn get_array_index_from_id(_cx: *mut JSContext, id: HandleId) -> Option<u32> {
+pub unsafe fn get_array_index_from_id(cx: *mut JSContext, id: HandleId) -> Option<u32> {
     let raw_id = id.into();
-    unsafe {
-        if RUST_JSID_IS_INT(raw_id) {
-            return Some(RUST_JSID_TO_INT(raw_id) as u32);
-        }
-        None
+    if RUST_JSID_IS_INT(raw_id) {
+        return Some(RUST_JSID_TO_INT(raw_id) as u32);
     }
-    // if id is length atom, -1, otherwise
-    // return if JSID_IS_ATOM(id) {
-    //     let atom = JSID_TO_ATOM(id);
-    //     //let s = *GetAtomChars(id);
-    //     if s > 'a' && s < 'z' {
-    //         return -1;
-    //     }
-    //
-    //     let i = 0;
-    //     let str = AtomToLinearString(JSID_TO_ATOM(id));
-    //     return if StringIsArray(str, &mut i) != 0 { i } else { -1 }
-    // } else {
-    //     IdToInt32(cx, id);
+
+    // if !RUST_JSID_IS_VOID(raw_id) {
```

Comment:
In the short term this could be `if raw_id.asBits == JSID_VOID.asBits` (based on https://doc.servo.org/mozjs_sys/jsid/constant.JSID_VOID.html). We can also add a RUST_JSID_IS_VOID API to https://github.com/servo/rust-mozjs/blob/master/src/glue.rs and https://github.com/servo/rust-mozjs/blob/master/src/jsglue.cpp.